pkg/config/file/file.go

```diff
@@ -30,3 +34,49 @@ func Decode(filePath string, element interface{}) error {
 
 	return parser.Fill(element, root, parser.FillerOpts{AllowSliceAsStruct: true})
 }
+
+// DecodeContent decodes the given configuration file content into the given element.
+// The operation goes through three stages roughly summarized as:
+// file contents -> tree of untyped nodes
+// untyped nodes -> nodes augmented with metadata such as kind (inferred from element)
+// "typed" nodes -> typed element.
+func DecodeContent(content string, extension string, element interface{}) error {
+	data := make(map[string]interface{})
+
+	switch extension {
+	case ".toml":
+		_, err := toml.Decode(content, &data)
+		if err != nil {
+			return err
+		}
+
+	case ".yml", ".yaml":
+		var err error
+		err = yaml.Unmarshal([]byte(content), &data)
+		if err != nil {
+			return err
+		}
+
+	default:
+		return fmt.Errorf("unsupported file extension: %s", extension)
+	}
+
+	filters := getRootFieldNames(element)
+
+	node, err := decodeRawToNode(data, parser.DefaultRootName, filters...)
+	if err != nil {
+		return err
+	}
+
+	if len(node.Children) == 0 {
+		return nil
+	}
+
+	metaOpts := parser.MetadataOpts{TagName: "noop", AllowSliceAsStruct: false}
```

Comment:
```suggestion
	metaOpts := parser.MetadataOpts{TagName: parser.TagLabel, AllowSliceAsStruct: false}
```

Does it makes sense to do something like this ?

Reply:
nope it's the expected tag name.